demucs/api.py

```diff
@@ -0,0 +1,834 @@
+
+        Returns
+        -------
+        A generator (iterator) of tuple[filename, wave]. If `ignore_errors` is True (default), the
+        wave of that file will be `None`.
+        """
+        for track in tracks:
+            if ignore_errors:
+                try:
+                    yield track, self._load_audio(track, audio_channels, samplerate)
+                except Exception:
+                    yield track, None
+            else:
+                yield track, self._load_audio(track, audio_channels, samplerate)
+
+    def load_audios_to_model(self, *tracks: Path):
+        """
+        Load several audios to the Separator that can be used in `Separator.separate_loaded_audio`.
+
+        Parameters
+        ----------
+        tracks: Pathlike objects, containing the path of the audio to be loaded.
+
+        Returns
+        -------
+        None
+
+        Notes
+        -----
+        When an error encountered, this function will only warn and continue loading other audios.
+        The audio failed to load will not be added into the Separator. To get a list of audios
+        failed to be loaded, you can use the following codes:
+        ```python
+        import warnings
+        with warnings.catch_warnings(record=True) as w:
+            Separator.load_audios_to_model(track)
+            failures = list(i.message.separate('"')[1] for i in w)
+        ```
+        """
+        if self._model is None:
+            raise RuntimeError("Please load model first!")
+        for track, wav in self.load_audios(*tracks):
+            if wav is None:
+                warnings.warn(f'"{track}" read failed and will not be loaded to model.')
+                continue
+            self._file.append(track)
+            self._wav.append(wav)
+
+    def load_audios_setup(self):
+        """
+        Load audios specied in the command line to the Separator that can be used in
+        `Separator.separate_loaded_audio`.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+
+        Notes
+        -----
+        When an error encountered, this function will only warn and continue loading other audios.
+        The audio failed to load will not be added into the Separator. To get a list of audios
+        failed to be loaded, you can use the following codes:
+        ```python
+        import warnings
+        with warnings.catch_warnings(record=True) as w:
+            Separator.load_audios_setup()
+            failures = list(i.message.separate('"')[1] for i in w)
+        ```
+        """
+        self.load_audios_to_model(*self._tracks)
+
+    def clear_filelist(self):
+        """
+        Remove all the loaded audios in the Separator.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+        self._wav = []
+        self._file = []
+
+    def add_track(self, filename: str, wav: th.FloatTensor):
```

Comment:
I would simplify the API, you do not really need to add tracks and then process them, as anyway the `apply_model` cannot work with more than one track at a time.
If I would have to describe the ideal API it would be something like that:
```
class Separator:
    def __init__(self, model and options args):
        ...
    def separate_audio_file(path, target_folder=None):...
         # this one will separate audio directly from the path and
         # to a target folder. If target folder is None, uses the same convention 
         # as the current commandline ?
    def separate_audio(audio: Tensor, samplerate: int) -> Tensor:
         # separate from a Tensor directly into a tensor.
```